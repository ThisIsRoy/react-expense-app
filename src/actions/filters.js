export const setTextFilter = (textFilter = '') => ({
  type: 'SET_TEXT_FILTER',
  text: textFilter
});

export const sortByDate = () => ({
  type: 'SORT_BY_DATE',
  sortBy: 'date'
});

export const sortByAmount = () => ({
  type: 'SORT_BY_AMOUNT',
  sortBy: 'amount'
});

export const setStartDate = (startDate = undefined) => ({
  type: 'SET_START_DATE',
  startDate: startDate
});

export const setEndDate = (endDate = undefined) => ({
  type: 'SET_END_DATE',
  endDate: endDate
});