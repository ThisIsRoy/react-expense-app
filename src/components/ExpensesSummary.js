import React from 'react';
import { connect } from 'react-redux';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummary = (props) => {
  const expensesCount = props.expenses.length;
  const expensesSum = numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00');

  if (expensesCount === 0) {
    return (<h1>There are no active expenses!</h1>);
  } else if (expensesCount === 1) {
    return (<h1>Viewing 1 expense totalling {expensesSum}</h1>);
  } else {
    return (<h1>Viewing {expensesCount} expenses totalling {expensesSum}</h1>);
  }
};

const mapStateToProps = (state) => ({
  expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpensesSummary);
