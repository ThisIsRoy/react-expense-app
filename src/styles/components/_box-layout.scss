.box-layout {
  align-items: center;
  background: url('/images/bg.jpg');
  background-size: cover;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100vw;
}

.box-layout__box {
  // background: fade-out(green, 0.15);
  color: white;
  text-align: center;
}

.login-text {
  animation: fadein 3s ease-in-out;
  //color: #79C753;
  color: white;
  font-size: $l-size;
  font-weight: 400;
  margin-bottom: $xl-size;
}

@keyframes fadein {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}