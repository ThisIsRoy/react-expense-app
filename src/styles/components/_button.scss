.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888888;
}

.login-button {
  background-color: #DC4C46;
  border: 1px solid white;
  border-radius: $s-size;
  color: white;
  font-size: $l-size;
  padding: $m-size $l-size;
  text-align: center;
  text-decoration: none;
  transition: 0.2s ease-in-out;
}

.login-button:hover {
  transform: scale(1.1);
}
