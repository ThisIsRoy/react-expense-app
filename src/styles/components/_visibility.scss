.show-mobile {
  font-weight: 700;
  @media (min-width: $desktop-breakpoint) {
    display: none;
  }
}

.show-desktop {
  font-size: $m-size;
  font-weight: 700;
  @media (max-width: $desktop-breakpoint - 0.01 rem) {
    display: none;
  }
}