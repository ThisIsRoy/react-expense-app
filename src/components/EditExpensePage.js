import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import expenses from '../selectors/expenses';
import { startEditExpense, startRemoveExpense } from '../actions/expense';

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };

  onClick = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
          <button className="button button--secondary" onClick={this.onClick}>Remove Expense</button>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find(eachExpense => eachExpense.id === props.match.params.id)
});

// removeExpense needs props as an input?
const mapDispatchToProps = (dispatch) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (expense) => dispatch(startRemoveExpense(expense))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);