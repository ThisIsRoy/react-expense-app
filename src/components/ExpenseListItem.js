import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import moment from 'moment';
import numeral from 'numeral';
import { startRemoveExpense } from '../actions/expense';

export class ExpenseListItem extends React.Component {
  constructor(props) {
    super(props);
  };

  onClick = (e) => {
    e.preventDefault();
    this.props.startRemoveExpense({ id: this.props.id });
    this.props.history.push('/');
  };

  render() {
    return (
      <Link className="list-item" to={"/edit/" + this.props.id}>
        <div>
          <h3 className="list-item__title">{this.props.description}</h3>
          <span className="list-item__subtitle">{numeral(this.props.amount / 100).format('$0,0.00')}</span>
        </div>
        <div>
          <h3 className="list-item__data">{moment(this.props.createdAt).format('MMMM Do, YYYY')}</h3>
          <button onClick={this.onClick} className="button--small">Remove</button>
        </div>
     </Link>
    );
  };
}

const mapDispatchToProps = (dispatch) => ({
  startRemoveExpense: (expense) => dispatch(startRemoveExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(ExpenseListItem);