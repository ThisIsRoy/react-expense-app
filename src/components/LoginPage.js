import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="login-text">A simple and elegant way of <br/>tracking your expenditures</h1>
      <button className="login-button" onClick={startLogin}>Log In</button>
    </div>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);