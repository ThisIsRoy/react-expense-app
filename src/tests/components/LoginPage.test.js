import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';
import { startLogin } from '../../actions/auth';

test('should render LoginPage correct', () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogout correctly on click', () => {
  const startLogin = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={startLogin} />);
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenLastCalledWith();
});