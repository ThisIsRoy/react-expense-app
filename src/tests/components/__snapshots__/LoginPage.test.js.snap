// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correct 1`] = `
<div>
  <p>
    This is the login page!
  </p>
  <button>
    Log In
  </button>
</div>
`;
